sdk/core/Azure.Core/src/TokenRequestContextOptions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Options for <see cref="TokenRequestContext"/> initialization.
+    /// </summary>
+    public struct TokenRequestContextOptions
```

Comment:
This type looks the same as `TokenRequestContext` just with setters. Considering `TokenRequestContext` is for internal use only I think we should stick to constructor parameters.

Reply:
The main reason for this is that the additional optional parameter causes lots of ambiguous ctor errors throughout the repo. I'm good with fixing them all, but this made it easier.